pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -207,9 +234,24 @@ func (ctrl *Controller) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer ctrl.queue.ShutDown()
 	defer ctrl.imgQueue.ShutDown()
+	listerCaches := []cache.InformerSynced{ctrl.mcpListerSynced, ctrl.mccrListerSynced, ctrl.ccListerSynced,
+		ctrl.imgListerSynced, ctrl.icspListerSynced, ctrl.idmsListerSynced, ctrl.itmsListerSynced, ctrl.clusterVersionListerSynced}
+
+	enabled, err := ctrl.sigstoreAPIEnabled()
+	if enabled {
+		err = ctrl.addImagePolicyObservers()
+		if err != nil {
+			klog.Infof("skipping error adding image policy observers: %v", err)
+		} else {
+			klog.Info("addded image policy observers")
+			ctrl.configInformerFactory.Start(stopCh)
+			listerCaches = append(listerCaches, ctrl.clusterImagePolicyListerSynced)
+		}
+	} else {
+		klog.Infof("skipping adding image policy observers with SigstoreImageVerification Featuregate value: %v, error: %v", enabled, err)
```

Comment:
Should this just ignore the error, or fail hard?

`runStartCmd` should have exited if the access fails, so if the query fails here, maybe we can exit instead of having to worry at all about what happens on this error path.

Reply:
Can not use Exit here, and if just return from here would silently fail the start of containerruntime config controller.:
```
 ./hack/golangci-lint.sh "containers_image_openpgp exclude_graphdriver_devicemapper exclude_graphdriver_btrfs containers_image_ostree_stub"
pkg/controller/container-runtime-config/container_runtime_config_controller.go:235:3: exitAfterDefer: os.Exit will exit, and `defer ctrl.imgQueue.ShutDown()` will not run (gocritic)
		os.Exit(1) 
```

It seems what we can do is just log out this error, so I only kept the log inside the `sigstoreAPIEnabled()`

Reply:
Fair enough; eliminating that risk would be nice, but it’s a purely theoretical risk, and adding an error reporting mechanism to `Run` does look rather out of scope of this PR.